src/snowflake/snowpark/_internal/analyzer/cte_utils.py

```diff
@@ -42,17 +47,20 @@ def find_duplicate_subtrees(root: "SnowflakePlan") -> Set["SnowflakePlan"]:
 
     This function is used to only include nodes that should be converted to CTEs.
     """
+    from snowflake.snowpark._internal.analyzer.select_statement import Selectable
+
     node_count_map = defaultdict(int)
     node_parents_map = defaultdict(set)
 
-    def traverse(node: "SnowflakePlan") -> None:
+    def traverse(node: "TreeNode") -> None:
         node_count_map[node] += 1
-        if node.source_plan and node.source_plan.children:
-            for child in node.source_plan.children:
-                node_parents_map[child].add(node)
-                traverse(child)
+        for child in node.children_plan_nodes:
+            if isinstance(child, Selectable):
+                child = child.to_subqueryable()
```

Comment:
It's required if the subquery is a non-select sql, otherwise this kind of test will fail https://github.com/snowflakedb/snowpark-python/blob/af8fbd78abfceeb404dd5d8acf73551edeb30147/tests/integ/scala/test_dataframe_suite.py#L425. I added a comment there